src/widget/wwidget.h

```diff
@@ -44,6 +45,21 @@ class WWidget : public QWidget, public WBaseWidget {
     ~WWidget() override;
 
     Q_PROPERTY(double value READ getControlParameterDisplay);
+    Q_PROPERTY(double backgroundColorRgba READ getBackgroundColorRgba WRITE
```

Comment:
I think we should not introduce another concept of background color for all widgets. 
We have already the background property and the BgColor xml element.

Reply:
To be honest, I'm not really familiar with how our skin system works. That code was written by @ferranpujolcamins. If you want to change it, could you please file a PR to my branch?

Reply:
Where is the background property?

```
$ grep -Hrn Q_PROPERTY src
src/track/track.h:56:    Q_PROPERTY(QString artist READ getArtist WRITE setArtist)
src/track/track.h:57:    Q_PROPERTY(QString title READ getTitle WRITE setTitle)
src/track/track.h:58:    Q_PROPERTY(QString album READ getAlbum WRITE setAlbum)
src/track/track.h:59:    Q_PROPERTY(QString albumArtist READ getAlbumArtist WRITE setAlbumArtist)
src/track/track.h:60:    Q_PROPERTY(QString genre READ getGenre WRITE setGenre)
src/track/track.h:61:    Q_PROPERTY(QString composer READ getComposer WRITE setComposer)
src/track/track.h:62:    Q_PROPERTY(QString grouping READ getGrouping WRITE setGrouping)
src/track/track.h:63:    Q_PROPERTY(QString year READ getYear WRITE setYear)
src/track/track.h:64:    Q_PROPERTY(QString track_number READ getTrackNumber WRITE setTrackNumber)
src/track/track.h:65:    Q_PROPERTY(QString track_total READ getTrackTotal WRITE setTrackTotal)
src/track/track.h:66:    Q_PROPERTY(int times_played READ getTimesPlayed)
src/track/track.h:67:    Q_PROPERTY(QString comment READ getComment WRITE setComment)
src/track/track.h:68:    Q_PROPERTY(double bpm READ getBpm WRITE setBpm)
src/track/track.h:69:    Q_PROPERTY(QString bpmFormatted READ getBpmText STORED false)
src/track/track.h:70:    Q_PROPERTY(QString key READ getKeyText WRITE setKeyText)
src/track/track.h:71:    Q_PROPERTY(double duration READ getDuration WRITE setDuration)
src/track/track.h:72:    Q_PROPERTY(QString durationFormatted READ getDurationTextSeconds STORED false)
src/track/track.h:73:    Q_PROPERTY(QString durationFormattedCentiseconds READ getDurationTextCentiseconds STORED false)
src/track/track.h:74:    Q_PROPERTY(QString durationFormattedMilliseconds READ getDurationTextMilliseconds STORED false)
src/track/track.h:131:    // Helper functions for Q_PROPERTYs
src/widget/wpushbutton.h:45:    Q_PROPERTY(bool pressed READ isPressed);
src/widget/wpushbutton.h:54:    Q_PROPERTY(bool hover READ isHovered);
src/widget/wpushbutton.h:64:    Q_PROPERTY(int displayValue READ readDisplayValue NOTIFY displayValueChanged)
src/widget/wwidget.h:47:    Q_PROPERTY(double value READ getControlParameterDisplay);
src/widget/wwidget.h:48:    Q_PROPERTY(double backgroundColorRgba READ getBackgroundColorRgba WRITE
src/widget/wwidget.h:50:    Q_PROPERTY(bool shouldHighlightBackgroundOnHover MEMBER
src/widget/wwidget.h:52:    Q_PROPERTY(bool hasBackgroundColor READ hasBackgroundColor);
src/widget/wwidget.h:53:    Q_PROPERTY(bool backgroundIsDark READ backgroundIsDark);
src/widget/wwidgetgroup.h:34:    Q_PROPERTY(int layoutSpacing READ layoutSpacing WRITE setLayoutSpacing DESIGNABLE true);
src/widget/wwidgetgroup.h:35:    Q_PROPERTY(QRect layoutContentsMargins READ layoutContentsMargins WRITE setLayoutContentsMargins DESIGNABLE true);
src/widget/wwidgetgroup.h:36:    Q_PROPERTY(Qt::Alignment layoutAlignment READ layoutAlignment WRITE setLayoutAlignment DESIGNABLE true);
src/widget/wwidgetgroup.h:57:    Q_PROPERTY(int highlight READ getHighlight WRITE setHighlight NOTIFY highlightChanged)
src/controllers/controllerengine.h:55:    Q_PROPERTY(QString id READ readId)
src/controllers/controllerengine.h:58:    //Q_PROPERTY(ConfigKey key READ key)
src/controllers/controllerengine.h:60:    //Q_PROPERTY(QScriptValue function READ function)
src/controllers/controllerengine.h:61:    Q_PROPERTY(bool isConnected READ readIsConnected)
```

Reply:
Its a internal property of QPushButton
https://doc.qt.io/archives/qt-4.8/stylesheet-reference.html

Reply:
I will do a PR